gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala

```diff
@@ -237,15 +237,7 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
     }
 
     transformChildren(child, basicScanExecTransformers)
-    basicScanExecTransformers.toSeq.map(
-      scan => {
-        scan match {
-          case op: BatchScanExecTransformer =>
-            op.asInstanceOf[BatchScanExecTransformer]
-          case op: FileSourceScanExecTransformer =>
-            op.asInstanceOf[FileSourceScanExecTransformer]
-        }
-      })
+    basicScanExecTransformers.toSeq
```

Comment:
Thanks for your work! Yes, this piece of code is removable.
I note the above code inside `transformChildren` can also be refactored, e.g., by removing asInstanceOf[TransformSupport] in Line#232 and using `case t: TransformSupport` instead. And it looks that two match blocks can be unified to one. Could you help do the refactor also?

Reply:
Yes, updated as suggestion